kernel/common/enumerable.rb

```diff
@@ -408,11 +419,19 @@ def drop_while
   end
 
   def each_cons(num)
-    return to_enum(:each_cons, num) unless block_given?
-
     n = Rubinius::Type.coerce_to_collection_index num
     raise ArgumentError, "invalid size: #{n}" if n <= 0
 
+    return to_enum(:each_cons, num) {
+      if (enum_size = enumerator_size).nil? 
```

Comment:
Ah wait, I think this means that if it _is_ a non-nil value it can be used i the other branches in the if statement. In that case I think it makes more sense to just use `enumerator_size` directly as using an assignment in an if statement is a bit confusing.